geode-core/src/main/java/org/apache/geode/cache/query/internal/NullToken.java

```diff
@@ -33,6 +33,7 @@
  */
 public class NullToken implements DataSerializableFixedID, Comparable {
 
+  @SuppressWarnings("lgtm[java/useless-null-check]")
```

Comment:
Since lgtm is not a "normal" warning to suppress, it may be worth explaining this somewhere why we are suppressing it (rather than fixing it).  Also it seems like we're using both `@SuppressWarnings("lgtm[])` and comments `// lgtm[]` should be using one consistently or is there certain use cases that require one or the other?

Reply:
Good call on explaining the suppression here and other places the annotation is used. As for the different approaches to suppressing, [LGTM requires that inline suppressions contain no line breaks](https://lgtm.com/help/lgtm/alert-suppression#where-in-code-do-i-add-suppression-comments), which would not be possible for some of the lines due to the spotless plugin automatically wrapping comments that extend too far to the right. In those cases, the @SuppressWarnings annotation is used, but is not used throughout because it would potentially mask other, legitimate alerts in methods that are currently present or that get introduced at a later date.

Reply:
Ok, the comments in the code help help.  I do wonder if we should make the link you posted available somewhere (like the readme or contributing.)

And as long as we put `@SuppressWarnings` at the lowest possible level that should prevent the accident masking of other suppressions so that looks good.

Reply:
I'll try to write up and add a section about LGTM alerts in the appropriate section of the wiki/docs.

I've tried to use the inline comment approach wherever possible, since that suppresses only the specified warning on the single line with the comment, so it's much more surgical than the annotation, which on a large method could have a fairly big blanket effect.